### DISCUSSION TITLE
Glava Configs!

### DISCUSSION THREAD

**Discussion by T-Crypt** [prasanthrangan/hyprdots#252] [Ideas]
Since we're pretty heavy on the audio side for these configs... 

What do you think about me adding some glava configs for an Audio Visualizer?

**Comment by prasanthrangan** [comment]
sure, how does the glava config file look? is it simple enough to generate on the fly for each theme?

**Comment by kRHYME7** [comment]
Can you please share what  are the pros and cons of cava and glava?

**Comment by akashgagda** [comment]
> Since we're pretty heavy on the audio side for these configs...
> 
> What do you think about me adding some glava configs for an Audio Visualizer?

hey bro is it still in WIP

**Reply by T-Crypt** [reply]
I'll work on this today - tomorrow! 

Should be able to use the colors from the wallbash file to generate the colors for the glava config

**Reply by akashgagda** [reply]
Thanks bro i will be waiting

**Reply by akashgagda** [reply]
any update bro

**Reply by kRHYME7** [reply]
Not very related but Have someone checked out the new plug in? Hyprwinwrap.

hyprwinwrap -> clone of xwinwrap, allows you to put any app as a wallpaper

**Reply by akashgagda** [reply]
No bro

**Reply by akashgagda** [reply]
@T-Crypt release glava config bro

**Reply by T-Crypt** [reply]
@akashgagda Im working on it to work with wallbash and using the new plugin hyprwinwrap. Issue is its still opening a window

**Reply by akashgagda** [reply]
> @akashgagda Im working on it to work with wallbash and using the new plugin hyprwinwrap. Issue is its still opening a window 

Bro any update?

**Reply by kRHYME7** [reply]
@akashgagda  Cava now has custom colors in case you want it.

**Reply by akashgagda** [reply]
Thank you

**Reply by kRHYME7** [reply]
> Issue is its still opening a window

We can run  GLava  by ```  nohup glava &    ```

![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/3ef71109-1cbc-4bbf-8ec5-1fbb9dce25b1)
![image](https://github.com/prasanthrangan/hyprdots/assets/53417443/cba4f022-f763-44c8-9cf7-eb116993435e)
```
plugin:hyprwinwrap {
        # class is an EXACT match and NOT a regex!
	class = GLava
    }
```

**Reply by se7en-x230** [reply]
That goes into hyprland.config?

**Reply by kRHYME7** [reply]
yup,

I think it only caters one class 

so we can simplify this into 
``` plugin:hyprwinwrap:class = GLava ``` 

Then maybe use a keybinding to launch it like ``` killall glava || glava ```

if you want to run it on terminal you can disown glava ``` nohup glava &   ```

**Reply by akashgagda** [reply]
Now it's glava time I'm so excited

**Reply by kRHYME7** [reply]
@akashgagda  

```  plugin:hyprwinwrap:class = GLava ```  in your ./userprefs.conf

Then maybe use a keybinding to launch it like ``` killall glava || glava ```

if you want to run it on terminal you can disown glava ```  nohup glava &  ```

**Comment by se7en-x230** [comment]
That's super cool
I will try tomorrow.
Thanks

**Comment by se7en-x230** [comment]
![240223_23h05m44s_screenshot](https://github.com/prasanthrangan/hyprdots/assets/69777713/d13bd64c-bcb5-4543-a163-769029430ee2)

**Comment by shandsy** [comment]
I don't get it - I added:

 plugin:hyprwinwrap:class = GLava to userprefs.conf

I added bind = $mainMod+Alt, M, exec, killall glava || nohup glava &

to my keybindings.conf

It still opens a window. Am I doing something wrong?

**Reply by shandsy** [reply]
Oh, and I also tried it without nohup, fwiw

**Reply by kRHYME7** [reply]
try to use another app that is sure to be running in Wayland. I think glava is running in xwayland here, No?

**Reply by shandsy** [reply]
I got it working with Xava at least. Does the job and looks nice, thanks! :)

**Comment by sitolam** [comment]
Will this be implemented in the project?
